### PR TITLE
add support for .deb .rpm files or package name specification

### PR DESCRIPTION
**Pull request by tod-uma**
I've added support for installing rpms and debs or using a package name. I've tried to follow what I could determine as conventions, but I was having a hard time finding why things were done in certain ways. I have tested it on as much as I can, but we don't really have a very diverse range of machines here. Also, I am not set in doing this in the ways that I did. If you have suggestions on how to do it better, please let me know.

### CODE REVIEW

**Comment by nwang92** [comment]
Conditionally approve, but given the number of changes I'll do some adhoc testing to verify things are working as intended

File: roles/splunk_common/tasks/get_facts.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@
   become: yes
   become_user: "{{ privileged_user }}"
   register: pre_existing_splunk_secret
+  become: yes
```
Comment (nwang92) [inline roles/splunk_common/tasks/get_facts.yml]: Getting some linting errors here due to duplicate keys in this stanza

Comment (nwang92) [reply roles/splunk_common/tasks/get_facts.yml]: You should also be able to run the checks that CircleCI is doing with linting by running `make lint` in the root of the repo

File: roles/splunk_common/tasks/get_facts_build_type.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+- name: Set splunk_build_type from variable
+  set_fact:
+    splunk_build_type: "{{ splunk.build_type }}"
+  when:
+    - "'build_type' in splunk"
+    - "splunk.build_type is not none"
+
+- name: Set splunk_build_type from build_location
```
Comment (jmeixensperger) [inline roles/splunk_common/tasks/get_facts_build_type.yml]: Can probably grab the suffix using in one play, but this is fine for now

File: roles/splunk_common/tasks/install_splunk_tgz.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    dest: "{{ splunk.opt }}"
+    owner: "{{ splunk.user }}"
+    group: "{{ splunk.group }}"
+    remote_src: "{{ splunk.build_remote_src }}"
+  register: install_result
+  until: install_result is succeeded
+  retries: "{{ retry_num }}"
+  delay: 3
+  become: yes
+  become_user: "{{ privileged_user }}"
```
Comment (jmeixensperger) [inline roles/splunk_common/tasks/install_splunk_tgz.yml]: Not sure if we should be installing as privileged user.

Comment (bb03) [reply roles/splunk_common/tasks/install_splunk_tgz.yml]: I think this is probably necessary on most systems, although for the linux ones we'll want to switch users.  We'll want to verify that the appropriate user is set downstream

**Comment by bb03** [comment]
@tod-uma Can you tell us a little bit about what systems you were able to test these changes on?  I want to get a sense of where to do testing

**Comment by tod-uma** [comment]
most of the systems I tested on are centos 7.

**Comment by bb03** [comment]
@tod-uma I took a look at this and found what may be some bugs (ran under debian).  I'll add specific comments, but I would like you to rerun and verify in your environment that the changes work for you

File: roles/splunk_common/tasks/get_facts_build_type.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  block:
+    #check if the build_location is a tgz
+    - name: "Set build_type (tgz)"
+      set_fact:
+        splunk_build_type: "tgz"
+      when: splunk.build_location is match(".*(\.tar\.gz|\.tgz)$")
+    - name: "Set build_type (tgz)"
```
Comment (bb03) [inline roles/splunk_common/tasks/get_facts_build_type.yml]: This should be named "Set build_type (msi)"

File: roles/splunk_common/tasks/get_facts_build_type.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        splunk_build_type: "rpm"
+      when: splunk.build_location is match(".*\.rpm$")
+    #check if the build_location is a deb
+    - name: "Set build_type (deb)"
+      set_fact:
+        splunk_build_type: "deb"
+      when: splunk.build_location is match(".*\.deb$")
+  when:
+    - splunk_build_type is not defined
+    - splunk_build_type is not none
```
Comment (bb03) [inline roles/splunk_common/tasks/get_facts_build_type.yml]: a third conditional should be added 
```- splunk.build_location is not none```

**Comment by tod-uma** [comment]
> @tod-uma I took a look at this and found what may be some bugs (ran under debian). I'll add specific comments, but I would like you to rerun and verify in your environment that the changes work for you

Sorry I haven't replied yet. We're in the process of upgrading our main splunk cluster from 6.6.8 to 7.3.x and it's been absorbing all of my time. I will hopefully be able to get back to this in a couple weeks.

File: roles/splunk_common/tasks/get_facts_build_type.yml (hunk 5 above)
Comment (jmeixensperger) [reply roles/splunk_common/tasks/get_facts_build_type.yml]: @bb03 do you want to add this condition as well?

**Pull request merged**